File: NQT/NumberSystem/decimal_to_binary.py
# convert a decimal number to binary without using built-in functions
decimal = 42
binary = ""
while decimal > 0:
    binary = str(decimal % 2) + binary
    decimal //= 2
print(binary)  # Output: 101010

# using built-in function
decimal = 42
binary = bin(decimal)[2:]  # Convert decimal to binary and remove '0b'
print(binary)  # Output: 101010

File: NQT/NumberSystem/Java/DecimalToBinary.java
package NumberSystem.Java;

public class DecimalToBinary {
    public static void main(String[] args) {
        int decimalNumber = 45; // You can change this value to test with different decimal numbers
        String binaryValue = decimalToBinary(decimalNumber);
        System.out.println("The binary value of decimal " + decimalNumber + " is: " + binaryValue);
    }

    private static String decimalToBinary(int decimal) {
        if (decimal == 0) {
            return "0";
        }
        StringBuilder binaryValue = new StringBuilder();
        while (decimal > 0) {
            int remainder = decimal % 2;
            binaryValue.insert(0, remainder); // Prepend the remainder to the binary string
            decimal /= 2; // Divide the decimal number by 2
        }
        return binaryValue.toString();
    }
}
